src/NuGet.Core/NuGet.Protocol/Resources/PackageUpdateResource.cs

```diff
@@ -242,27 +261,54 @@ public Uri SourceUri
                     packagePath));
             }
 
-            var sourceUri = UriUtility.CreateSourceUri(source);
+            Uri packageSourceUri = UriUtility.CreateSourceUri(source);
 
-            if (string.IsNullOrEmpty(apiKey) && !sourceUri.IsFile)
+            if (string.IsNullOrEmpty(apiKey) && !packageSourceUri.IsFile)
             {
                 log.LogWarning(string.Format(CultureInfo.CurrentCulture,
                     Strings.NoApiKeyFound,
                     GetSourceDisplayName(source)));
             }
 
-            await PushAll(source, apiKey, noServiceEndpoint, skipDuplicate, requestTimeout, log, packagesToPush: nupkgsToPush, token);
-        }
-
-        private async Task PushAll(string source, string apiKey, bool noServiceEndpoint, bool skipDuplicate, TimeSpan requestTimeout, ILogger log, IEnumerable<string> packagesToPush, CancellationToken token)
-        {
-            foreach (var packageToPush in packagesToPush)
+            foreach (string nupkgToPush in nupkgsToPush)
             {
-                await PushPackageCore(source, apiKey, packageToPush, noServiceEndpoint, skipDuplicate, requestTimeout, log, token);
+                bool packageWasPushed = await PushPackageCore(source, apiKey, nupkgToPush, noServiceEndpoint, skipDuplicate, requestTimeout, log, token);
```

Comment:
We need to know whether this nupkg was actually pushed, so we can proceed with its symbol package.
Previously, this happened in 2 separate batches. The idea here was to push the symbol immediately iff the nupkg succeeded, rather than "remembering" which nupkgs were successful.